src/lib/translators/mongo.ts

```diff
@@ -1340,6 +1340,78 @@ function transformTop(step: Readonly<S.TopStep>): MongoStep[] {
   ];
 }
 
+/** transform a 'totals' step into corresponding mongo pipeline steps */
+function transformTotals(step: Readonly<S.AddTotalRowsStep>): MongoStep {
+  const facet: { [id: string]: MongoStep[] } = {};
+  const groups: string[] = step.groups ?? [];
+  const addFields: MongoStep = {};
+  const project: MongoStep = { _id: 0 }; // ensures $project stage will never be empty
+  // list of columns to combine
+  const toCombine: string[] = [...step.totals.map(c => c[0])];
+  // get combinations, remove first combo (most granular combination of columns
+  // so not useful to compute total rows) and add[](to compute the grand total)
+  const combos: string[][] = combinations(toCombine).slice(1);
+  combos.push([]);
+
+  //Keep original data in first facet
+  for (const aggfStep of step.aggregations) {
+    for (let i = 0; i < aggfStep.columns.length; i++) {
+      addFields[aggfStep.newcolumns[i]] = $$(aggfStep.columns[i]);
+      if (aggfStep.newcolumns[i] !== aggfStep.columns[i]) {
+        project[aggfStep.columns[i]] = 0;
+      }
+    }
+  }
+  facet['originalData'] = [{ $addFields: addFields }, { $project: project }];
+
+  for (let i = 0; i < combos.length; i++) {
+    const comb = combos[i];
+    const id = columnMap([...comb, ...groups]);
```

Comment:
```suggestion
   // List of columns identifying groups we're gonna compute the total of
    const id = columnMap([...comb, ...groups]);
```

Reply:
(am I right? I'm not entirely sure)

Reply:
I've opted for:

`
// List of columns that that will be used to group the aggregations computation
// i.e. we will compute total rows for dimensions not included in this group id
`

(It's actually the opposite of what you suggested which means it was not clear, sorry!)